lib/device-log/ios-log.js

```diff
@@ -249,7 +267,33 @@ class IOSLog {
   async getAllLogs () {
     return this.logs;
   }
+
+  async killExistingSysLogProcesses (cmd, args) {
+    if (!IOSLog.cachedIDeviceSysLogs[this.subprocessId]) {
+      await exec('pkill', ['-xf', [cmd, ...args].join(' ')]);
+    }
+  }
+
+  /**
+   * Get 'idevicesyslog' subprocess for a given udid.
+   *
+   * This function caches and re-uses instances so that there's
+   * just one instance per device (cache by UDID)
+   */
+  getIDeviceSysLogProcess (cmd, args, env) {
+    if (!IOSLog.cachedIDeviceSysLogs[this.subprocessId]) {
+      IOSLog.cachedIDeviceSysLogs[this.subprocessId] = {
+        proc: new SubProcess(cmd, args, {env}),
+        count: 1,
+      };
+    } else {
+      IOSLog.cachedIDeviceSysLogs[this.subprocessId].count++;
```

Comment:
this counter should not be increased if the process exists in the cache, but is not running, although it is not checked

Reply:
@mykola-mokhnach Is there ever a case where the process wouldn't be running?

Reply:
yep, the process might die if the device has been unexpectedly disconnected or it has been killed manually or just because of a bug

Reply:
Maybe something like

```javascript
proc.on('exit', () => delete IOSLog.cachedIDeviceSysLogs[this.subprocessId]);
```

be a good approach?

Reply:
@mykola-mokhnach I added a fix for this.